crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -61,20 +62,30 @@ use crate::{
     },
     storage::{SparseSetIndex, TableId, TableRow},
 };
-use core::{fmt, hash::Hash, mem, num::NonZero, sync::atomic::Ordering};
+use core::{fmt, hash::Hash, mem, num::NonZero};
 #[cfg(feature = "serialize")]
 use serde::{Deserialize, Serialize};
 
-#[cfg(target_has_atomic = "64")]
+#[cfg(not(feature = "portable-atomic"))]
+use core::sync::atomic::Ordering;
+
+#[cfg(feature = "portable-atomic")]
+use portable_atomic::Ordering;
+
+#[cfg(all(target_has_atomic = "64", not(feature = "portable-atomic")))]
 use core::sync::atomic::AtomicI64 as AtomicIdCursor;
+#[cfg(all(target_has_atomic = "64", feature = "portable-atomic"))]
+use portable_atomic::AtomicI64 as AtomicIdCursor;
 #[cfg(target_has_atomic = "64")]
 type IdCursor = i64;
 
 /// Most modern platforms support 64-bit atomics, but some less-common platforms
 /// do not. This fallback allows compilation using a 32-bit cursor instead, with
 /// the caveat that some conversions may fail (and panic) at runtime.
-#[cfg(not(target_has_atomic = "64"))]
+#[cfg(all(not(target_has_atomic = "64"), not(feature = "portable-atomic")))]
 use core::sync::atomic::AtomicIsize as AtomicIdCursor;
+#[cfg(all(not(target_has_atomic = "64"), feature = "portable-atomic"))]
+use portable_atomic::AtomicIsize as AtomicIdCursor;
```

Comment:
I haven't tried it, but I believe using `portable-atomic` may allow access to an `AtomicI64` on all platforms, removing the possible panic. This should be investigated in a follow-up PR.